GVRf/Framework/framework/src/main/jni/vulkan/vulkanCore.cpp

```diff
@@ -1334,12 +1337,106 @@ void VulkanCore::InitPipelineForRenderData(const GVR_VK_Vertices* m_vertices, Vu
     }
 
     VulkanCore::~VulkanCore() {
-        vkDestroyDevice(getDevice(), nullptr);
+
+        //destroy all the resources associated with the current device.
```

Comment:
I don't think this ever gets called. See issue https://github.com/Samsung/GearVRf/issues/1535.